tests/test_ERC20.py

```diff
@@ -1,41 +1,194 @@
+    # Check if the user is approved
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == approved_amount
+
+@pytest.mark.asyncio
+async def test_approve_none_minted():
+    _, erc20, owner, friend = await erc20_factory()
+    amount = uint(69420)
+    await erc20.approve(friend.contract_address, amount).invoke(owner.contract_address)
+    # Check if the user is approved
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == amount
+    executed_info = await erc20.allowance(friend.contract_address, owner.contract_address).call()
+    assert executed_info.result.allowance == uint(0)
+
+@pytest.mark.asyncio
+async def test_fail_approve_zero_spender():
+    _, erc20, owner, _ = await erc20_factory()
+    amount = uint(69420)
+    await erc20.mint(owner.contract_address, amount).invoke(caller_address=owner.contract_address)
+    await assert_invoked_revert(erc20.approve(0, amount), owner.contract_address)
+
+##########################
+## increase_allowance() ##
+##########################
+
+@pytest.mark.asyncio
+async def test_increase_allowance():
+    _, erc20, owner, friend = await erc20_factory()
+    half = uint(500)
+    amount = uint(1000)
+    # Approve the user to spend the tokens
+    await erc20.approve(friend.contract_address, half).invoke(caller_address=owner.contract_address)
+    # Check if the user is approved to spend
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == half
+    executed_info = await erc20.allowance(friend.contract_address, owner.contract_address).call()
+    assert executed_info.result.allowance == uint(0)
+    # Increase Allowance
+    executed = await erc20.increase_allowance(friend.contract_address, half).invoke(caller_address=owner.contract_address)
+    assert executed.result.success == 1
+    # Check increased allowance
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == amount
+    executed_info = await erc20.allowance(friend.contract_address, owner.contract_address).call()
+    assert executed_info.result.allowance == uint(0)
+
+@pytest.mark.asyncio
+async def test_increase_allowance_overflow():
+    _, erc20, owner, friend = await erc20_factory()
+    amount = MAX_UINT256
+    # overflow_amount adds (1, 0) to (2**128 - 1, 2**128 - 1)
+    overflow_amount = uint(1)
+    # Mint
+    await erc20.mint(owner.contract_address, amount).invoke(caller_address=owner.contract_address)
+    # Approve
+    await erc20.approve(friend.contract_address, amount).invoke(caller_address=owner.contract_address)
+    # Overflow
+    await assert_invoked_revert(erc20.increase_allowance(friend.contract_address, amount), owner.contract_address)
```

Comment:
Not quite sure as I am not super familiar with pytest, but apparently the prefered method of expecting errors is

```
with pytest.raises(Exception):
    await erc20.increase_allowance(friend.contract_address, amount).invoke()
```